src/main/java/com/blazemeter/ciworkflow/CiBuild.java

```diff
@@ -35,7 +36,7 @@
 
     public String pr;
 
-    public CiBuild(AbstractTest test, String properties,
+    public CiBuild(AbstractTest test, JSONArray properties,
```

Comment:
Can we convert Properties from String to JSONArray in some public static method that will be call from constructor?

Reply:
I think this should be done inside of particular CI system(Jenkins/TeamCity/Bamboo) before passing to CiBuild, because we also need to resolve environment variables(if there are any in `properties`). And each CI system has its own mechanism of resolving. So I think better to move this substitution out of this project and to pass to CiBuild constructor resolved JSONArray.

Reply:
But you can resolve environment variables String properties before set in this class or add special resolve() method that user can override.